core/src/main/java/nl/nn/adapterframework/stream/MessageContext.java

```diff
@@ -42,24 +50,63 @@ public class MessageContext extends LinkedHashMap<String,Object> {
 	public static final String METADATA_LOCATION = "Metadata.Location";
 	public static final String METADATA_MIMETYPE = "Metadata.MimeType";
 
+	private Map<String, Serializable> data = new LinkedHashMap<>();
+
 	public MessageContext() {
 		super();
 	}
 	public MessageContext(String charset) {
 		this();
 		withCharset(charset);
 	}
-	public MessageContext(Map<String,Object> base) {
+	public MessageContext(Map<String, ?> base) {
 		this();
 		withAllFrom(base);
 	}
 
-	public MessageContext withAllFrom(Map<String,Object> base) {
+	public MessageContext(MessageContext base) {
+		this(base.data);
+	}
```

Comment:
Is dit handig? Wanneer wordt dit gebruikt? Misschien dat het handiger is om een copy methode te hebben. Anders hebben we straks mogelijk 2 pointers naar de zelfde context.

Reply:
De data wordt wel gekopieerd in die andere constructor, dus we hebben niet 2 pointers naar dezelfde. Ik zal even opzoeken waar het gebruikt wordt, om te kijken of het nodig / wenselijk is.

Reply:
Deze constructor wordt 2x gebruikt, in `Message.copyContext()` en in de constructor van `PathMessage`.
De constructor die een map als parameter neemt wordt ook 2x gebruikt: in `FileSystemUtils.getContext()` en in de constructor van `UrlMessage`.

Ik zou deze constructors kunnen weghalen en vervangen door `.withAllFrom(...)`?